autopilot/scanner/scanner.go

```diff
@@ -82,18 +85,32 @@ func New(hs HostStore, scanBatchSize, scanThreads uint64, scanMinInterval time.D
 
 		statsHostPingMS: utils.NewDataPoints(0),
 		logger:          logger.Sugar(),
-
-		interruptChan: make(chan struct{}),
-		shutdownChan:  make(chan struct{}),
 	}, nil
 }
 
 func (s *scanner) Scan(ctx context.Context, hs HostScanner, force bool) {
-	if s.canSkipScan(force) {
+	s.mu.Lock()
+	if force {
+		if s.scanning {
```

Comment:
I think this check is redundant now. We can close the context and wait regardless of whether we are scanning.